File: mall-dao/src/main/java/cn/dnaizn/mall/service/SellerCatService.java
package cn.dnaizn.mall.service;

import entity.PageResult;
import cn.dnaizn.mall.pojo.SellerCat;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SellerCatService {

	/**
	 * 返回全部列表
	 * @return
	 */
	List<SellerCat> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	void add(SellerCat sellerCat);
	
	
	/**
	 * 修改
	 */
	void update(SellerCat sellerCat);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	SellerCat findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	PageResult findPage(SellerCat sellerCat, int pageNum, int pageSize);

	/**
	 *
	 * @param parentId
	 * @return
	 */
	List<SellerCat> findByParentId(Long parentId);
}

File: mall-manager-web/src/main/resources/static/js/service/sellerCatService.js
//服务层
app.service('sellerCatService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../sellerCat/findAll');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../sellerCat/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../sellerCat/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../sellerCat/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../sellerCat/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../sellerCat/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../sellerCat/search?page='+page+"&rows="+rows, searchEntity);
	}
	this.findByParentId=function(parentId){
		return $http.get('../sellerCat/findByParentId?parentId='+parentId);
	}
});
